eradicate.py

```diff
@@ -28,176 +28,199 @@
 import io
 import os
 import re
-import sys
 import tokenize
 
 __version__ = '1.0'
 
+class Eradicator:
+    """Eradicate comments."""
+    MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
+    PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
+
+    DEFAULT_WHITELIST = [
+        r'pylint',
+        r'pyright',
+        r'noqa',
+        r'type:\s*ignore',
+        r'fmt:\s*(on|off)',
+        r'TODO',
+        r'FIXME',
+        r'XXX'
+    ]
+    WHITELIST_REGEX = re.compile(r'|'.join(DEFAULT_WHITELIST), flags=re.IGNORECASE)
+
+    def comment_contains_code(self, line, aggressive=True):
+        """Return True comment contains code."""
+        line = line.lstrip()
+        if not line.startswith('#'):
+            return False
+
+        line = line.lstrip(' \t\v\n#').strip()
+
+        # Ignore non-comment related hashes. For example, "# Issue #999".
+        if re.search('#[0-9]', line):
+            return False
+
+        # Ignore whitelisted comments
+        if self.WHITELIST_REGEX.search(line):
+            return False
+
+        if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
```

Comment:
Let's move all `re` constants to a single place.

Reply:
I named it `CODING_COMMENT_REGEX` .. dunno what kind of comments this regex should catch. I don't use a tool which has `coding` comments.

Do you have an idea or better variable name?

Reply:
This is all about `# coding: utf8` comments from `python2`
Your name is fine!

Reply:
Ah .. now that you say it I remember